subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultComponentMetadataHandler.java

```diff
@@ -230,6 +309,26 @@ public boolean isSatisfiedBy(ComponentMetadataDetails componentMetadataDetails)
         }
     }
 
+    static class ComponentMetadataContextMatchingSpec implements Spec<ComponentMetadataContext> {
+        private ModuleIdentifier target;
+
+        ComponentMetadataContextMatchingSpec(ModuleIdentifier target) {
+            this.target = target;
+        }
+
+        public boolean isSatisfiedBy(ComponentMetadataContext componentMetadataContext) {
+            return componentMetadataContext.getDetails().getId().getGroup().equals(target.getGroup()) && componentMetadataContext.getDetails().getId().getName().equals(target.getName());
```

Comment:
:o: Would read better by extracting the id in a local variable.

Reply:
Done